### ISSUE TITLE
skip_if_bz_open & bug coverage

### ISSUE THREAD

**Issue by rochacbruno** [SatelliteQE/robottelo#3419]
Some random thoughts about the skip_if_bz decorator. I was thinking if it is a good idea when we remove it. 

I mean, if a test has a related Bz it should have a persistent relationship so with that we can in future do a "Bug Coverage Report" for automation. 

That analysis can be useful to realize which features of product is missing tests and which tests makes a real sense.

When we remove the skip_if_bz we lose the relation and history. 

Wouldn't be a better idea to have a "@related_bzs("123", "456", "789", ...)"  or even putting this on docstring?

And also in some centralized place, maybe a file "skip.conf" we put all the skipping BZs, and then all tests related to that Bzs is skipped.

Just a random idea because all sources of data is worth to keep, we never know the value it will have in the future.

And when you go to a test you can clearly see if there is a BZ explaining why that test exists, vice-verse, you can get a BZ number and see which tests are covering that issue, sometimes it avoids the creation of more tests with the same purpose also helps if the bug is a regression.

**Comment by elyezer** [comment]
Testimony have the `@bz` tag which we can list the bugs, currently we are adding it whenever we decorate with a bug and removing when removing the decorator. Maybe we can leave the docstring bit and use testimony to generate that report to us.

The point will be, if that the test is blocked by that bz or it is just related. To this I have no idea on how to solve.

I'm not sure on creating a new decorator for related bugs because that will require a lot of work when a test is first skipped by a bug, it will require 1) add the skip decorator and list the BZ number, 2) add the testimony tag and list the BZ number and 3) add the related decorator and list the BZ number. I know this is the worse case but it happened a lot of time these past weeks.

**Comment by sthirugn** [comment]
To make it simple, we can just enter the bugzilla info as first line of the function code.  Something like:

```
def test_negative_reimport_config_files(self):
        """Repetitive Import of all Config Files from default
        @feature: Repetitive Import Config Files
        @assert: All Config Files are imported only once
        """
        # See bz 1234567, 7654321, 12345
        # See redmine 12345, 54321, 56789
        for test_data in gen_import_config_files_data():
            with self.subTest(test_data):
            ...
            ...
```

We dont need to put this info on docstring because it is not required for any kind of reporting.

**Comment by rochacbruno** [comment]
Closing, as it looks like this issue is not relevant or already covered by Polarion.
